server/src/service-layer/controllers/BookingController.ts

```diff
@@ -32,6 +35,94 @@ import {
 
 @Route("bookings")
 export class BookingController extends Controller {
+  @SuccessResponse("200", "Bookings successfully created")
+  @Security("jwt", ["admin"])
+  @Post("create-bookings")
+  public async createBookings(
+    @Body() requestBody: BookingsByDateRangeRequestModel
+  ): Promise<UIdssByDateRangeResponse> {
+    try {
+      const { startDate, endDate } = requestBody
+
+      /** Creating instances of the required services */
+      const bookingSlotService = new BookingSlotService()
+      const bookingDataService = new BookingDataService()
+
+      // Query to get all booking slots within date range
+      const bookingSlots =
+        await bookingSlotService.getBookingSlotsBetweenDateRange(
+          startDate,
+          endDate
+        )
+
+      /** The response data array */
+      const responseData: Array<{
+        date: Timestamp
+        users: string[]
+      }> = []
+
+      /** Iterating through each booking slot */
+      const bookingPromises = bookingSlots.map(async (slot) => {
+        /** Getting the bookings for the current slot */
+        const bookings = await bookingDataService.getBookingsBySlotId(slot.id)
+
+        /** Extracting the all 3 Ids from the bookings */
+        const userIds = bookings.map((booking) => booking.user_id)
+        const slotIds = bookings.map((booking) => booking.booking_slot_id)
+        const stripePaymentIds = bookings.map(
+          (booking) => booking.stripe_payment_id
+        )
+
+        if (userIds.length === 0) {
+          return
+        }
+
+        /** For every slotid add a booking for that id only if user doesn't already have a booking */
+        let i: number
+        for (i = 0; i < userIds.length; i++) {
+          if (
+            (await bookingDataService.getBookingsByUserId(userIds[i]))
+              .length !== 0
+          ) {
+            delete userIds[i] // Remove user from list if they already have a booking
+            break
+          } else {
+            bookingDataService.createBooking({
+              user_id: userIds[i],
+              booking_slot_id: slotIds[i],
+              stripe_payment_id: stripePaymentIds[i]
+            })
+          }
+        }
```

Comment:
refer to #459, this will get very slow for large data sets, nede to use `Promise.all`